mantidproject/mantid
# Add flood correction to RROA

**Description of work.**
Call ApplyFloodWorkspace from RROA for the input and transmission workspaces.

**To test:**
1. Create a flood workspace using CreateFloodWorkspace algorithm. Use it with RROA, set `FloodCorrection` property to "Workspace".
2. Set `FloodCorrection` property to "ParameterFile". Add flood correction parameters to the appropriate parameter file. The parameters should have the form:
```
  <parameter name="Flood_Run" type="string">
    <value val=" FILENAME_OR_RUN_NUMBER(S)" />
  </parameter>

  <parameter name="Flood_PROPERTYNAME" type="string">
    <value val="PROPERTYVALUE" />
  </parameter>
```
where `PROPERTYNAME` is a property name of `CreateFloodWorkspace` algorithm (excluding Filename and OutputWorkspace).

Fixes #23483

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
